### ISSUE TITLE
strict rule does not apply to function expressions

### ISSUE THREAD

**Issue by martin-probst** [eslint/eslint#1244]
Function expressions like

```
var foo = function() {
  return "bar";
};
```

do not trigger the **strict** rule warning 'missing "use strict" statement' while it does work with a function declaration like

```
function foo() {
  return "bar";
}
```
